### ISSUE TITLE
Startup Jenkins with custom certs added to keystore

### ISSUE THREAD

**Issue by BogdanSorlea** [jenkinsci/kubernetes-operator#192]
Hey,

I'm trying to configure Jenkins to work with our internal LDAP installation, to which it connects via `ldaps://`. This is achieved using the Jenkins LDAP plugin - https://wiki.jenkins.io/display/JENKINS/LDAP+Plugin

Now, because the cert for our LDAP master is signed by our own CA, the cert chain (I guess) would need to be installed/configured "in" Jenkins' JVM/JRE via some keystore - https://stackoverflow.com/questions/12066430/javax-naming-communicationexception-simple-bind-failed

So I guess I could in principle create a keystore using `keytool` command and I could instruct Jenkins to take into account by passing parameters like here - https://support.cloudbees.com/hc/en-us/articles/203821254-How-to-install-a-new-SSL-certificate/#3addthecertificatetothejenkinsstartupparameters - via the JAVA_ARGS or JAVA_OPTS env var (not entirely sure which, let me know if I'm off), env var added to the Jenkins helm chart under `spec/master/containers/{jenkins-master}/env` map.

What is unclear to me is how to best pass the keystore file to the container - is there some way to create the file as some sort of value, via helm?

I am not sure how to go about in this matter, so some help/examples could help.

Thanks

LE: I tried to import the resulting cacerts file as file read from ConfigMap (written as binary64-encoded value) - but upon decoding and creating the value kubernetes fucks something up, as the two files are not the same, which results in a corrupt keystore file and corresponding errors on LDAP test:
```
Unable to connect to ldaps://<whatever> : javax.naming.CommunicationException: <whatever>:636 [Root exception is java.net.SocketException: java.security.NoSuchAlgorithmException: Error constructing implementation (algorithm: Default, provider: SunJSSE, class: sun.security.ssl.SSLContextImpl$DefaultSSLContext)]
```
```
There is a problem with reading default keystore/truststore file /var/jenkins/home/.cacerts/cacerts with the passphrase changeit - the file won't be loaded. The reason is: Invalid keystore format
[...]
The attempt #1 to do the action check updates server failed with an allowed exception:
java.io.IOException: Invalid keystore format
    [...]
Caused: java.security.KeyStoreException: problem accessing trust store
```

MD5 hashes of the two files, as well as their size are (very different):
```
$ md5 -s cacerts.out
MD5 ("cacerts.out") = ca0f807da054a1e9f3a5e18114ee22e1
$ md5 -s cacerts.fromjenkins
MD5 ("cacerts.fromjenkins") = f6275aeb69827fa19e178316641a0706
$ ll
-rw-r--r--    1 me   staff  176178 Nov 11 18:35 cacerts.out
-rw-r--r--    1 root  staff  272804 Nov 11 18:40 cacerts.fromjenkins
```

I even tried (manual) base64 encoding, followed by decoding, followed by validating the converted file (via `keytool -list -v -keystore <filename>` command) and it works - it is only after it passes through kubernetes that it becomes corrupted somehow. What's interesting is that the beginning of the files seems to be the same, although upon diff it doesn't, which suggests maybe some file encoding error, somehow?
```
$ head -n2 cacerts.out > head.out
$ head -n2 cacerts.fromjenkins > head.jenkins
$ diff head.out head.jenkins
Binary files head.out and head.jenkins differ
$ cat head.out
0��1�4ve*�H��nclass2publicprimarycertificationauthority-g3m�>�XX.5090�0�ap�I�_�E)簦�P[z0
    0	UUS10U
VeriSign, Inc.10U
                 VeriSign Trust Network1:08U
                                            1(c) 1999 VeriSign, Inc. - For authorized us360716235959Z0��1ign Class 2 Public Primary Certification Authority - G30
                 0	UUS10U
$ cat head.jenkins
0��1�4ve*�H��nclass2publicprimarycertificationauthority-g3m�>�XX.5090�0�ap�I�_�E)簦�P[z0
    0	UUS10U
VeriSign, Inc.10U
                 VeriSign Trust Network1:08U
                                            1(c) 1999 VeriSign, Inc. - For authorized us360716235959Z0��1ign Class 2 Public Primary Certification Authority - G30
                 0	UUS10U
```

I have added the value into a ConfigMap as a `base64 -b 80`-encoded value, with the content of the file defined as:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: cacerts
data:
  cacerts: !!binary |
    [(base 64 -b 80)-value here]
```

So I'm not sure which way to go

**Comment by tomaszsek** [comment]
Hi @BogdanSorlea 

First of all I recommend use Kubernetes secret to store the trust store. You can try with command `kubectl create secret generic trust-store --from-file=./truststore`. Can you provide your Jenkins CR?

Cheers

**Comment by BogdanSorlea** [comment]
This is a visual diff of the beginning of the two files mentioned above:
<img width="638" alt="Screen Shot 2019-11-11 at 20 12 51" src="https://user-images.githubusercontent.com/4004804/68613842-f0985080-04bf-11ea-8bf5-3a82c6543d48.png">

I will get back with my Jenkins CR config, can't post right now

**Comment by BogdanSorlea** [comment]
@tomaszsek I did a quick test based on what you suggested above and it seems to work! Thanks a lot.

I still need to figure out how I could put in code this trust-store secret, if possible. If you have any idea, please let me know

**Comment by tomaszsek** [comment]
Yep my bad for only CAs config map will be enough. In current state of the operator helm chart you can't add config maps. There are two options:
- modify the helm chart to add ability to provision config maps(contributions are more than welcome https://github.com/jenkinsci/kubernetes-operator/tree/master/chart/jenkins-operator)
- create separate config map yaml file and apply through `kubectl apply -f file.yaml`

**Comment by pawelprazak** [comment]
you can also create a "wrapper" chart, you could put your custom config in there and add the jenkins operator helm chart as a dependency

this should make it easier to compose it

**Comment by tomaszsek** [comment]
@BogdanSorlea Do you still have the problem?

**Comment by tomaszsek** [comment]
In the new version of the helm chart, config maps with configurations will be created for you.

**Comment by tomaszsek** [comment]
Fixed in `0.1.0` version of Helm Chart.
